File: Add Code Here/LeetCode Problems/1 Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = []  # Initialize an empty list to store the indices of the two numbers

        # Iterate through the elements of the array
        for i in range(0, len(nums)):
            for j in range(i + 1, len(nums)):  # Nested loop to check pairs
                if nums[i] + nums[j] == target:  # Check if the pair sums to the target
                    l.append(i)  # Add the first index to the list
                    l.append(j)  # Add the second index to the list

        return l  # Return the list containing the indices of the two numbers that add up to the target

File: Add Code Here/LeetCode Problems/1-Two-sum.js
var twoSum = function (nums, target) {
	let map = new Map();

	// console.log(map);

	for (var i = 0; i < nums.length; i++) {
		let complement = target - nums[i];

		console.log("complement's" + complement);
		if (map.has(complement)) {
			return [map.get(complement), i];
		}
		map.set(nums[i], i);
	}

	console.log("after" + map);
};

console.log(twoSum([7, 4, 1, 3], 5));

var twoSum = function (nums, target) {
	let length = nums.length;
	let currentIndex = 0;
	for (let i = 1; i <= length && length != 1; i++) {
		if (i != length && nums[currentIndex] + nums[currentIndex + i] == target) {
			console.log(`[${currentIndex},${currentIndex + i}]`);
			return [currentIndex, currentIndex + i];
		} else if (i == length) {
			i = 0;
			length--;
			currentIndex++;
		}
	}
};
